batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/hyperlink/BatchPropertyHyperlinkDetector.java

```diff
@@ -0,0 +1,150 @@
+			return null;
+		
+		if(textEditor.getEditorInput() instanceof IFileEditorInput){
+			file = ((IFileEditorInput)textEditor.getEditorInput()).getFile();
+		}
+		
+		int offset= region.getOffset();
+		
+		ITypeRoot input= EditorUtility.getEditorInputJavaElement(textEditor, true);
+		if (input == null)
+			return null;
+
+		IDocument document= textEditor.getDocumentProvider().getDocument(textEditor.getEditorInput());
+		IRegion wordRegion= JavaWordFinder.findWord(document, offset);
+		if (wordRegion == null)
+			return null;
+		
+		IProject project = null;
+		
+		project = input.getJavaProject().getProject();
+		
+		if(project == null)
+			return null;
+		
+		BatchProject batchProject = (BatchProject) BatchProjectFactory.getBatchProjectWithProgress(project);
+
+		if (batchProject == null) {
+			return null;
+		}
+		
+		IJavaElement[] elements = null;
+		
+		try {
+			elements = input.codeSelect(wordRegion.getOffset(), wordRegion.getLength());
+			
+			if (elements == null) 
+				return null;
+			
+			if(elements.length != 1)
+				return null;
+			
+			ArrayList<IHyperlink> hyperlinks = new ArrayList<IHyperlink>();
+			
+			IField field = null;
+			
+			if(elements[0] instanceof IType){
+				elements[0] = input.getElementAt(wordRegion.getOffset());
+				
+				if(elements[0] == null)
+					return null;
+				
+				if(elements[0] instanceof IField){
+					field = (IField) elements[0];
+				}
+			}else if(elements[0] instanceof IField){
+				field = (IField) elements[0];
+			}
+			
+			if(field != null){
+				IType type = field.getDeclaringType();
+				IBatchArtifact artifact = batchProject.getArtifact(type);
+				if (artifact != null) {
+					for (IBatchProperty property : artifact.getProperties()) {
+						if (property.getField().equals(field)) {
+							hyperlinks.add(new BatchPropertyHyperlink(region, property, document));
+						}
+					}
+				}
+			}
+			if (hyperlinks != null && !hyperlinks.isEmpty()) {
+				return (IHyperlink[])hyperlinks.toArray(new IHyperlink[hyperlinks.size()]);
+			}
+		} catch (JavaModelException jme) {
+			BatchUIPlugin.getDefault().logError(jme);
+		}
+		return null;
+	}
+
+	private IRegion getAttributeValueSourceRegion(ITextViewer textViewer, int offset, Attr attribute) {
+		IRegion sourceRegion = null;
+		try {
+			sourceRegion = Utils.getAttributeValueRegion(textViewer.getDocument(), attribute);
+		} catch (BadLocationException e) {
+			BatchUIPlugin.getDefault().logError(e);
+		}
+
+		if (sourceRegion != null && offset >= sourceRegion.getOffset()
+				&& offset <= (sourceRegion.getOffset() + sourceRegion.getLength())) {
+			return sourceRegion;
+		}
+		return null;
+	}
+
+	private IBatchProject getBatchProject(IFile file) {
+		IBatchProject batchProject = BatchProjectFactory.getBatchProjectWithProgress(file.getProject());
+
+		if (batchProject != null) {
+			for (IFile f : batchProject.getDeclaredBatchJobs()) {
+				if (f.equals(file)) {
+					return batchProject;
+				}
+			}
+		}
```

Comment:
Scanning can be replaced with

if(batchProject != null && batchProject.getDeclaredBatchJobs().contains(file)) {
	return batchProject;
}